### ISSUE TITLE
Зона ответственности Repository

### ISSUE THREAD

**Issue by YuryYuroppi** [Faidien/Planner#1]
У вас класс Repository - это и швец и жнец и на дуде игрец,  надо как то разделить ответственность.
Какое требование вы могли бы отнести к классу Repository ?

**Comment by Faidien** [comment]
Думаю все, что непосредственно относится к репозиторию - создание массива записей, доступ к ним, выгрузка\загрузка.

**Comment by Faidien** [comment]
> У вас класс Repository - это и швец и жнец и на дуде игрец, надо как то разделить ответственность.
> Какое требование вы могли бы отнести к классу Repository ?

Думаю все, что непосредственно относится к репозиторию - создание массива записей, доступ к ним, выгрузка\загрузка.

**Comment by YuryYuroppi** [comment]
Класс Repository отвечает за хранение, добавление, поиск, сортировку и удаление заметок. Все. 
Почему он отвечает за ввод-вывод и работу с консолью ? 
Наше требование к нему - "хранение, добавление, поиск, сортировку и удаление заметок".
В соответствии Single res. principle - наш класс должен меняться только в случае изменения данного требования. Для данного класса - это его единственная причина для изменения.
Окей, теперь мы используем данный класс в проекте ASP.NET Core - нам нужен консольный ввод ? Требование к вводу изменилось и наш класс уже придется переписывать. Это вторая причина для изменения (а она должна быть одна).
Используем вместо диска - передачу по сети, все - меняем класс. Это наша третья причина для изменения.

**Comment by Faidien** [comment]
> Окей, теперь мы используем данный класс в проекте ASP.NET Core - нам нужен консольный ввод ? Требование к вводу изменилось и наш класс уже придется переписывать. Это вторая причина для изменения (а она должна быть одна).

То есть, все Console... надо заменить на использование ссылки на метод в доп классе?

**Comment by YuryYuroppi** [comment]
Пусть за работу с пользователем отвечает класс меню. А за работу с диском что-нибудь вроде FileRecordSource, реализующий IRecordSource.

**Comment by YuryYuroppi** [comment]
> То есть, все Console... надо заменить на использование ссылки на метод в доп классе?

Есть класс Меню. Меню отвечает за одну вещь. За обработку команд с консоли.
Пользователь попросил что то сделать, например добавить заметку. Меню, используя класс Repository - добавит заметку и обработает необходимый ввод вывод.

IRecordSource пусть содержит 2 метода - загрузить и сохранить.

**Comment by YuryYuroppi** [comment]
Да, еще - замените использование массива на List<T> - сократит код.
public void CreateBase(string path) - данного метода не должно существовать. Давайте рассмотрим его сигнатуру CreateBase(string path) - во первых непонятно что это значит. Во вторых документация к методу некорректна - данный метод создает файл в любом случае, хотя написано, что он создает его, если его нет. 
Далее - вы используете string как параметр, это не совсем то что нужно - хотите подставить путь ? Используйте тип Path.

Методы Add, Resize удаляем после использования List<T>.
